packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import classNames from 'classnames/bind';
+import IndexExampleTemplate from './IndexPageTemplate';
+import styles from './DocTemplate.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The given component's npm package name
+   */
+  packageName: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * All of the example(s) that will be displayed. An empty array is supported.
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * All of the props table(s) that will be displayed. An empty array is supported.
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  packageName: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
+
+const DocTemplate = ({ packageName, readme, examples, propsTables, ...customProps }) => {
+  let id = 0;
+  const localExamples = examples;
+  const localPropsTables = propsTables;
+
+  // Assign unique identifiers to use as keys later
+  for (let i = 0; i < localExamples.length; i += 1) {
+    localExamples[i].id = id;
+    id += 1;
+  }
+
+  for (let i = 0; i < localPropsTables.length; i += 1) {
+    localPropsTables[i].id = id;
+    id += 1;
+  }
+
+  const version = `[![NPM version](http://img.shields.io/npm/v/${packageName}.svg)](https://www.npmjs.org/package/${packageName})`;
+
+  return (
+    <div {...customProps}>
+      {packageName && <Markdown src={version} />}
```

Comment:
What should we do if version isn't present? If packageName isn't present, that could mean version potentially isn't as well.

Reply:
`version` is no longer a prop and is just the markdown to display npm's version image for the package. It's always present.